frontend/javascripts/oxalis/model/sagas/proofread_saga.ts

```diff
@@ -309,8 +309,9 @@ function* splitOrMergeOrMinCutAgglomerate(
   yield* call([Model, Model.ensureSavedState]);
 
   /* Reload the segmentation */
-
-  yield* call([api.data, api.data.reloadBuckets], layerName);
+  yield* call([api.data, api.data.reloadBuckets], layerName, (bucket) =>
+    bucket.containsValue(targetAgglomerateId),
```

Comment:
@fm3 Can you confirm that only voxels that have the the target agglomerate id will change after merge/split/min-cut? My tests confirmed this, but I want to doublecheck with you that this is really deterministic.

Reply:
It is deterministic in the backend. I am not familiar with the term target agglomerate id.

Split only affects one “before” agglomerate so that should be safe. Min-Cut itself does not have side effects, instead, the front-end uses its return value to then issue a sequence of splits.

The merge udpate actions have agglomerateId1 and agglomerateId2. The segments that were agglomerateId2 will be mapped to agglomerateId1 instead. So if the frontend’s targetAgglomerateId is equal to the update action’s agglomerateId2, this should be safe.

Reply:
great, I think, everything should work then :+1: